lib/ansible/modules/cloud/vmware/vmware_guest_tools_wait.py

```diff
@@ -0,0 +1,204 @@
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_guest_tools_wait
+short_description: Wait for VMware tools to become available and return facts
+description:
+    - Wait for VMware tools to become available on the VM and return facts.
+version_added: '2.4'
+author:
+    - Philippe Dellaert (@pdellaert) <philippe@dellaert.org>
+notes:
+    - Tested on vSphere 6.5
+requirements:
+    - python >= 2.6
+    - PyVmomi
+options:
+   name:
+        description:
+            - Name of the VM for which to wait until the tools become available.
+            - This is required if uuid is not supplied.
+   name_match:
+        description:
+            - If multiple VMs match the name, use the first or last found.
+        default: 'first'
+        choices: ['first', 'last']
+   folder:
+        description:
+            - Destination folder, absolute or relative path to find an existing guest.
+            - This is required if C(name) is supplied.
+            - The folder should include the datacenter. ESX's datacenter is C(ha-datacenter).
+            - 'Examples:'
+            - '   folder: /ha-datacenter/vm'
+            - '   folder: ha-datacenter/vm'
+            - '   folder: /datacenter1/vm'
+            - '   folder: datacenter1/vm'
+            - '   folder: /datacenter1/vm/folder1'
+            - '   folder: datacenter1/vm/folder1'
+            - '   folder: /folder1/datacenter1/vm'
+            - '   folder: folder1/datacenter1/vm'
+            - '   folder: /folder1/datacenter1/vm/folder2'
+        default: /vm
+   uuid:
+        description:
+            - UUID of the VM  for which to wait until the tools become available, if known. This is VMware's unique identifier.
+            - This is required if C(name) is not supplied.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Wait for VMware tools to become available by UUID
+  vmware_guest_tools_wait:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: no
+    uuid: 421e4592-c069-924d-ce20-7e7533fab926
+  delegate_to: localhost
+  register: facts
+
+- name: Wait for VMware tools to become available by name
+  vmware_guest_tools_wait:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: no
+    name: test-vm
+    folder: /datacenter1/vm
+  delegate_to: localhost
+  register: facts
+'''
+
+RETURN = """
+instance:
+    description: metadata about the virtual machine
+    returned: always
+    type: dict
+    sample: None
+"""
+
+import time
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.vmware import connect_to_api, gather_vm_facts, vmware_argument_spec, find_vm_by_id
+
+try:
+    import json
```

Comment:
I learned yesterday that nowadays we can simply do `import json` as we no longer support Python 2.4